lib/network/rest/server.js

```diff
@@ -0,0 +1,332 @@
+'use strict'; // eslint-disable-line
+
+const httpServer = require('../http/server');
+const Logger = require('werelogs').Logger;
+const url = require('url');
+const assert = require('assert');
+
+const constants = require('../../constants');
+const utils = require('./utils');
+const errors = require('../../errors');
+
+/**
+ * Parse the Range header into a two-integers array with possibly
+ * undefined values
+ *
+ * @param {String} rangeHeader - The 'Range' header value
+ * @return {(Number|undefined) []} An array of [left, right] integer
+ *   values provided in the range specification if present, or
+ *   undefined otherwise.
+ */
+function parseRangeSpec(rangeHeader) {
```

Comment:
I prefer having separate functions for separate concerns of 1. parsing the range spec and 2. getting the absolute byte range to read from the object.

Also `parseRange` looks a bit too complex and error-prone for what it does, and it's in S3 code (which would mean moving it to arsenal).

Though I agree it's not ideal to have duplication. Ideally (to me!) we would have both functions in Arsenal, used by both the data server and S3 server. Do you think it's worth doing this change?

Overall, those two functions are small and (hopefully) well tested so I'm not too concerned about keeping as-is.